Orange/widgets/visualize/widgetutils/tree/rules.py

```diff
@@ -0,0 +1,181 @@
+class Rule:
+    """The base Rule class for tree rules."""
+
+    def merge_with(self, rule):
+        """Merge the current rule with the given rule.
+
+        Parameters
+        ----------
+        rule : Rule
+
+        Returns
+        -------
+        Rule
+
+        """
+        raise NotImplemented()
+
+
+class DiscreteRule(Rule):
+    """Discrete rule class for handling Indicator rules.
+
+    Parameters
+    ----------
+    attr_name : str
+    eq : bool
+        Should indicate whether or not the rule equals the value or not.
+    value : object
+
+    Examples
+    --------
+    Age = 30
+    >>> rule = DiscreteRule('age', True, 30)
+    Name ≠ John
+    >>> rule = DiscreteRule('name', False, 'John')
+
+    Notes
+    -----
+      - Merging discrete rules is currently not implemented, the new rule is
+        simply returned and a warning is printed to stderr.
+
+    """
+
+    def __init__(self, attr_name, eq, value):
+        self.attr_name = attr_name
+        self.sign = eq
+        self.value = value
+
+    def merge_with(self, rule):
+        # It does not make sense to merge discrete rules, since they can only
+        # be eq or not eq.
+        from sys import stderr
+        print('WARNING: Merged two discrete rules `%s` and `%s`'
+              % (self, rule), file=stderr)
+        return rule
+
+    def __str__(self):
+        return '{} {} {}'.format(
+            self.attr_name, '=' if self.sign else '≠', self.value)
+
+
+class ContinuousRule(Rule):
+    """Continuous rule class for handling numeric rules.
+
+    Parameters
+    ----------
+    attr_name : str
+    gt : bool
+        Should indicate whether the variable must be greater than the value.
+    value : int
+    inclusive : bool, optional
+        Should the variable range include the value or not
+        (LT <> LTE | GT <> GTE). Default is False.
+
+    Examples
+    --------
+    x ≤ 30
+    >>> rule = ContinuousRule('age', False, 30, inclusive=True)
+    x > 30
+    >>> rule = ContinuousRule('age', True, 30)
+
+    Notes
+    -----
+      - Continuous rules can currently only be merged with other continuous
+        rules.
+
+    """
+
+    def __init__(self, attr_name, gt, value, inclusive=False):
+        self.attr_name = attr_name
+        self.sign = gt
```

Comment:
The reason I left it as `sign` is in order to keep some consistency between the subclasses of `Rule`.

Seeing as I did never actually use this general property and that it will most likely cause nothing but confusion, I will change the `sign` in the `ContinuousRule` to `gt` and `sign` in `DiscreteRule` to `eq`.